test/KillSwitchOracleIntegration.t.sol

```diff
@@ -0,0 +1,285 @@
+import { IACLManager }          from "lib/aave-v3-core/contracts/interfaces/IACLManager.sol";
+import { IPool }                from "lib/aave-v3-core/contracts/interfaces/IPool.sol";
+import { IPoolConfigurator }    from "lib/aave-v3-core/contracts/interfaces/IPoolConfigurator.sol";
+import { ReserveConfiguration } from "lib/aave-v3-core/contracts/protocol/libraries/configuration/ReserveConfiguration.sol";
+import { DataTypes }            from "lib/aave-v3-core/contracts/protocol/libraries/types/DataTypes.sol";
+
+import { KillSwitchOracle } from "src/KillSwitchOracle.sol";
+
+contract KillSwitchOracleIntegrationTest is Test {
+
+    using ReserveConfiguration for DataTypes.ReserveConfigurationMap;
+
+    event Trigger(address indexed oracle, uint256 threshold, uint256 price);
+    event AssetLTV0(address indexed asset);
+    event AssetFrozen(address indexed asset);
+
+    address constant POOL              = 0xC13e21B648A5Ee794902342038FF3aDAB66BE987;
+    address constant POOL_CONFIGURATOR = 0x542DBa469bdE58FAeE189ffB60C6b49CE60E0738;
+    address constant ACL_MANAGER       = 0xdA135Cd78A086025BcdC87B038a1C462032b510C;
+    address constant SPARK_PROXY       = 0x3300f198988e4C9C63F75dF86De36421f06af8c4;
+
+    address constant WBTC_ORACLE  = 0xfdFD9C85aD200c506Cf9e21F1FD8dd01932FBB23;
+    address constant STETH_ORACLE = 0x86392dC19c0b719886221c78AB11eb8Cf5c52812;
+
+    address constant DAI    = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address constant SDAI   = 0x83F20F44975D03b1b09e64809B757c47f942BEeA;
+    address constant USDC   = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address constant WETH   = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address constant WSTETH = 0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0;
+    address constant WBTC   = 0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599;
+    address constant GNO    = 0x6810e776880C02933D47DB1b9fc05908e5386b96;
+    address constant RETH   = 0xae78736Cd615f374D3085123A210448E74Fc6393;
+    address constant USDT   = 0xdAC17F958D2ee523a2206206994597C13D831ec7;
+
+    IPool             pool             = IPool(POOL);
+    IPoolConfigurator poolConfigurator = IPoolConfigurator(POOL_CONFIGURATOR);
+    IACLManager       aclManager       = IACLManager(ACL_MANAGER);
+
+    address[] assets;
+
+    address randomUser = makeAddr("randomUser");
+
+    KillSwitchOracle public killSwitchOracle;
+
+    function setUp() public {
+        vm.createSelectFork(getChain("mainnet").rpcUrl, 19255277);  // Feb 18, 2024
+
+        killSwitchOracle = new KillSwitchOracle(pool, poolConfigurator);
+        killSwitchOracle.transferOwnership(SPARK_PROXY);
+
+        vm.prank(SPARK_PROXY);
+        aclManager.addRiskAdmin(address(killSwitchOracle));
+
+        assets = pool.getReservesList();
+
+        // Add oracles for the pegged assets
+        // TODO: want to add reth too, but need to divide by exchange rate
```

Comment:
So this would wait for a separately developed adapter, right?

Reply:
Yeah we need to build this first. It should be a separate PR.